### PR TITLE
fix: include auto-association work in `gas` cost of transfer system contracts

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - C.f. #14234 
 - Increases `gas` cost of a transfer system contract by how many auto-associations are reported in the stream builder returned from dispatch.
 - Covered in `@HapiTest` that does NFT transfer with and without auto-association and asserts the USD price of the increased gas cost is $0.06 (HAPI cost with 20% surcharge through EVM).

### CODE REVIEW

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/extensions/SpecEntityExtension.java [hunk 1]
```diff
@@ -155,7 +155,9 @@ private SpecAccount accountFrom(@Nullable final AccountSpec annotation, @NonNull
                 .map(AccountSpec::name)
                 .filter(n -> !n.isBlank())
                 .orElse(defaultName);
-        return new SpecAccount(name);
+        final var account = new SpecAccount(name);
+        account.builder().maxAutoAssociations(annotation == null ? 0 : annotation.maxAutoAssociations());
```
Comment (david-bakin-sl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/extensions/SpecEntityExtension.java:159]: Isn't this statement a noop?

**Review by david-bakin-sl — COMMENTED** [review]
Have a question.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/extensions/SpecEntityExtension.java (hunk 1 above)
Comment (david-bakin-sl) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/extensions/SpecEntityExtension.java:159]: NVM, I get it.  Atypical use of "builder" pattern here.

**Review by lukelee-sl — APPROVED** [review]
LGTM.

**Review by lukelee-sl — APPROVED** [review]
LGTM

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM  from `hedera-base`

**Comment by tinker-michaelj** [comment]
Not needed in `v0.52` since unlimited auto-associations were disabled.

**Pull request closed without merging**